core/shared/src/main/scala/matryoshka/package.scala

```diff
@@ -655,6 +655,15 @@ package object matryoshka extends CofreeInstances with FreeInstances {
   def toTree[F[_]: Functor: Foldable]: Algebra[F, Tree[F[Unit]]] =
     x => Tree.Node(x.void, x.toStream)
 
+  /** Replaces all instances of `original` in the structure with `replacement`.
+    *
+    * @group algebras
+    */
+  def substitute[T[_[_]], F[_]](original: T[F], replacement: T[F])(implicit T: Equal[T[F]]):
+      T[F] => T[F] \/ T[F] =
+   tf => if (tf ≟ original) replacement.left else tf.right
```

Comment:
Interesting. Isn’t the former akin to `find(a ≟ b)`? Although, I don’t know how the latter works (i.e., the types are too common).

`orig: T[F] => replacement: T[F] => maybeReplacement: Option[T[F]]`

`orig: T[F] => maybeReplacement: Option[T[F]] => repOrOrig: T[F] \/ T[F]`

So the latter is really `toLeft` then, correct?

Hrmm … shouldn’t the first function just be `orig: T[F] => Option[T[F]]` (or `(T[F] => Boolean) => Option[T[F]]`) then the latter is confusing – Because if the option is `Some`, then we want to throw away its value and use the replacement, otherwise we want to use the unmodifed value.

Anyway, yeah, I dunno what the right decomposition of this is.